### PR TITLE
feat: Update settings loader to use db instead of json file

### PR DESCRIPTION
**Pull request by liordino**

### CODE REVIEW

File: lib/new-settings-loader.js [hunk 1]
```diff
@@ -1,71 +1,115 @@
...
-  return new Promise((resolve) => {
-    if (!db) {
-      setTimeout(() => {
-        return resolve(db.getState())
-      }, 1000)
-    } else {
-      return resolve(db.getState())
-    }
-  })
+  if (!versionId) throw new Error('versionId is required')
```
Comment (RafaelTaranto) [inline lib/new-settings-loader.js]: Better to change this to Promise.reject

File: lib/new-settings-loader.js [hunk 2]
```diff
@@ -1,71 +1,115 @@
...
-const adapter = new FileAsync('db.json')
-let db = null
-
-// TODO new-admin save to actual db, like the old settings
-low(adapter).then(it => {
-  db = it
-})
+const NEW_SETTINGS_LOADER_SCHEMA_VERSION = 2
 
+// TODO: validate
```
Comment (RafaelTaranto) [inline lib/new-settings-loader.js]: We don't want to leave TODOs on our code base, is there a reason for these here? What needs to be validated?

If this is referring to the "valid" row we won't need it anymore, new admin should work in a way that the user never inputs a invalid config.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: lib/new-settings-loader.js (hunk 2 above)
Comment (liordino) [reply lib/new-settings-loader.js]: Yeah, it refers to a valid row. I'll remove that, shouldn't be commited.

File: lib/new-settings-loader.js [hunk 3]
```diff
@@ -1,71 +1,111 @@
...
+      return db.none(sql, ['accounts', { accounts: newAccounts }, true, NEW_SETTINGS_LOADER_SCHEMA_VERSION])
+    })
 }
 
-function getAccounts () {
-  const state = db.getState()
-  return state ? state.accounts : null
+function loadAccounts () {
+  const toFields = fieldArr => _.fromPairs(_.map(r => [r.code, r.value], fieldArr))
+  const toPairs = r => [r.code, toFields(r.fields)]
```
Comment (RafaelTaranto) [inline lib/new-settings-loader.js]: Why is this needed? I don't see why it was working before now it's required.

File: lib/new-settings-loader.js [hunk 4]
```diff
@@ -1,71 +1,111 @@
...
+  const sql = `select data
+  from user_config
+  where type=$1
+  and schema_version=$2
+  and valid
+  order by id desc
+  limit 1`
+
+  return db.oneOrNone(sql, ['config', NEW_SETTINGS_LOADER_SCHEMA_VERSION])
+    .then(row => !row ? {} : row.data.config)
```
Comment (RafaelTaranto) [inline lib/new-settings-loader.js]: Is it possible for the query to return a null response? Isn't this QueryResultError already?

Also we don't want to return a empty object for config, just throw the error if we dont find a config.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Comment by liordino** [comment]
[Update settings loader to use db instead of json file](https://trello.com/c/33AkxRRN/153-update-settings-loader-to-use-db-instead-of-json-file)

**Review by RafaelTaranto — APPROVED** [review]

File: migrations/1595295132361-schema-version-on-user-config.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const db = require('./db')
+
+module.exports.up = function (next) {
+  var sql = [
+    'alter table user_config add column schema_version smallint DEFAULT 1'
```
Comment (RafaelTaranto) [inline migrations/1595295132361-schema-version-on-user-config.js]: Ahh, this should not be nullable.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]
